File: leetcode2/2medium/김대원/Q677.java
class MapSum {

    private static Set<String> keys;
    private static Map<String, Integer> map;

    public MapSum() {
        keys = new HashSet<>();
        map = new HashMap<>();
    }

    public void insert(String key, int val) {
        map.put(key, val);
        keys.add(key);
    }

    public int sum(String prefix) {
        int sums = 0;
        for (String key : keys) {
            if (key.startsWith(prefix)) sums += map.get(key);
        }

        return sums;
    }
}

File: leetcode2/2medium/강병호/Q677.py
class MapSum:

    def __init__(self):
        self.map_data = dict() 
        
    def insert(self, key: str, val: int) -> None:
        self.map_data[key] = val

    def sum(self, prefix: str) -> int:
        key_list = list(self.map_data.keys())
        result = 0

        for key in key_list:
            if prefix == key[0:len(prefix)]:
                result += self.map_data[key]
            
        
        return result
        


# Your MapSum object will be instantiated and called as such:
# obj = MapSum()
# obj.insert(key,val)
# param_2 = obj.sum(prefix)